jax/numpy/lax_numpy.py

```diff
@@ -394,7 +394,11 @@ def fn(x1, x2):
 bitwise_and = _one_to_one_binop(np.bitwise_and, lax.bitwise_and)
 bitwise_or = _one_to_one_binop(np.bitwise_or, lax.bitwise_or)
 bitwise_xor = _one_to_one_binop(np.bitwise_xor, lax.bitwise_xor)
-right_shift = _one_to_one_binop(np.right_shift, lax.shift_right_arithmetic)
+@_wraps(np.right_shift)
+def right_shift(x1, x2):
+  lax_fn = lax.shift_right_logical if \
+    np.issubdtype(x1.dtype, np.unsignedinteger) else lax.shift_right_arithmetic
+  return lax_fn(*_promote_args(np.right_shift.__name__, x1, x2))
```

Comment:
Also, I think the fork on dtype should probably happen after the argument promotion, otherwise you might get unexpected results when x1 is unsigned and x2 is signed.

Reply:
Done!